internal/pkg/service/appsproxy/dataapps/api/config.go

```diff
@@ -56,11 +57,15 @@ func (c AppConfig) IdAndName() string {
 	return c.Name + "-" + c.ID.String()
 }
 
+func (c AppConfig) Slug() string {
+	return *c.AppSlug
+}
+
 func (c AppConfig) Domain() string {
-	if c.Name == "" {
+	if c.AppSlug == nil {
```

Comment:
`if c.AppSlug == nil || *c.AppSlug == ""`, to prevent `-12345` prefix if the value is empty string